core-bundle/src/Routing/PageFinder.php

```diff
@@ -23,9 +24,28 @@ class PageFinder
     public function __construct(
         private readonly ContaoFramework $framework,
         private readonly RequestMatcherInterface $requestMatcher,
+        private readonly RequestStack|null $requestStack = null,
```

Comment:
```suggestion
    /**
     * @internal
     */
    public function __construct(
        private readonly ContaoFramework $framework,
        private readonly RequestMatcherInterface $requestMatcher,
        private readonly RequestStack $requestStack,
```
According to our BC promise: _“Our BC promise does not cover classes and methods marked as `@internal`. In most cases this concerns constructors of services Contao provides. If you want to change the behaviour of a service, do not replace it with your own instance of the class but instead decorate the original service”_

Reply:
Changed in 8f17837e23a3b5a700316e39696c8114831173b6.